### DISCUSSION TITLE
Question:Can the execution order of registered tags be specified?

### DISCUSSION THREAD

**Discussion by openai03** [harttle/liquidjs#760] [Q&A] [answered]
In Shopify's Liquid files, some default configuration items are read from the default values in the {% schema %}...{% endschema %} section at the end of the file, but the tag parsing is done in order. When the registered schema tag is called, the previous value-fetching process has already ended. So is there a way to specify that the schema I registered is executed first?
`like this
  <div class="banner__media animate--{{ section.settings.image_behavior }}"></div>
 {%  schema %}
{
      "type": "select",
      "id": "image_behavior",
      "default": "none",
      "label": "t:sections.all.animation.image_behavior.label"
    },
{% endschema%}
`

**Comment by harttle** [comment]
Not possible as the parsing is one time process apart from recursions.
But you can save it to somewhere during parse of ‘schema’ and read the values during render of other tags. But only your own tags can read the values.
If the values need to be added to ctx and the schema is always at the end, maybe you can transform the file before passed to liquidjs.

**Reply by openai03** [reply, marked as answer]
"Thank you, I have solved it. After calling parseFile, I obtained a token with the name 'schema'. Since the content in schema is generally a JSON-compliant object, I just need to extract the content from schema, then use JSON.parse on it, and finally put it into the context for rendering."
